### PR TITLE
Fix ticking weeks issue in batch edit

### PR DESCRIPTION
**Pull request by Coenego**
Ticking one week should always result in ticking **one** week, exactly the one that the user has clicked on :), the only time it is OK to automatically tick anything that the user has clicked on, is when the combination of days + weeks would result in an OT event

### CODE REVIEW

**Comment by Coenego** [comment]
Assigning to @bp323 for review

**Comment by bp323** [comment]
The same problem still seems to persist. Try the following using the day selector:
- Batch edit with one day, default is Tuesday so let's start from that
- Change that day to a Thursday in batch edit --> note that the checked week jumps one up
- Change that day to a Friday in batch edit --> note that it jumps again
- etc. etc.

Or try doing the following, only using the week tickboxes:
- Batch edit with one day, default is Tuesday so let's start from that
- Add another day and make it a Thursday
- Now click another week --> the week gets ticket but also the week after that. It should only add events to the ticked week.

**Comment by Coenego** [comment]
Back to @bp323 for review

File: shared/gh/js/views/gh.admin-batch-edit-date.js [hunk 1]
```diff
@@ -169,44 +169,55 @@ define(['lodash', 'moment', 'gh.core', 'gh.api.config'], function(_, moment, gh,
             // For each selected week, add another day
             _.each($weeks, function(chk) {
                 _.defer(function() {
+
                     // Get the week number
                     var eventWeek = parseInt(chk.value, 10);
 
                     // Get the date by week and day
                     var dateByWeekAndDay = gh.utils.getDateByWeekAndDay(termName, eventWeek, dayOfTheWeek);
+                    if (!_.contains([2,3], dayOfTheWeek)) {
```
Comment (bp323) [inline shared/gh/js/views/gh.admin-batch-edit-date.js]: Can you add a line of comments here?

File: shared/gh/js/views/gh.admin-batch-edit-date.js [hunk 2]
```diff
@@ -169,44 +169,55 @@ define(['lodash', 'moment', 'gh.core', 'gh.api.config'], function(_, moment, gh,
...
-                            'end': gh.utils.convertUnixDatetoISODate(moment(endDate).toISOString()),
-                            'location': defaultLocation,
-                            'type': gh.config.events.default,
-                            'organisers': defaultOrganisers,
-                            'start': gh.utils.convertUnixDatetoISODate(moment(startDate).toISOString())
-                        },
-                        'startDate': startDate
-                    });
+                    // Only add a new event for the generated day if the day is within a term
+                    if (gh.utils.getTerm(gh.utils.convertISODatetoUnixDate(moment(dateByWeekAndDay).utc().format()), true)) {
```
Comment (bp323) [inline shared/gh/js/views/gh.admin-batch-edit-date.js]: OT events are still being generated, or am I misreading this comment?

File: shared/gh/js/views/gh.admin-batch-edit-date.js [hunk 3]
```diff
@@ -275,10 +287,15 @@ define(['lodash', 'moment', 'gh.core', 'gh.api.config'], function(_, moment, gh,
                             var eventDay = parseInt($('.gh-batch-edit-day-picker', $timePickerContainer).val(), 10);
                             // Get the date by week and day
                             var dateByWeekAndDay = gh.utils.getDateByWeekAndDay(termName, eventWeek, eventDay);
+                            if (!_.contains([2,3], eventDay)) {
```
Comment (bp323) [inline shared/gh/js/views/gh.admin-batch-edit-date.js]: Same here, can you add a line of comments?

File: shared/gh/js/views/gh.admin-batch-edit-date.js [hunk 4]
```diff
@@ -495,32 +512,44 @@ define(['lodash', 'moment', 'gh.core', 'gh.api.config'], function(_, moment, gh,
...
-                    var eventEnd = new Date($row.find('.gh-event-date').attr('data-end'));
+                    var eventEnd = moment($row.find('.gh-event-date').attr('data-end'));
                     // Get which week of the term this event takes place in
-                    var weekInTerm = gh.utils.getAcademicWeekNumber(gh.utils.convertISODatetoUnixDate(moment(eventStart).toISOString()));
+                    var weekInTerm = gh.utils.getAcademicWeekNumber(gh.utils.convertISODatetoUnixDate(moment(eventStart).utc().format()));
                     // Get the name of the term this date is in
                     var termName = $row.closest('.gh-batch-edit-events-container').attr('data-term');
                     // Get the date the event would be on after the change
                     var newDate = gh.utils.getDateByWeekAndDay(termName, weekInTerm, eventDay);
+                    if (!_.contains([2,3], eventDay)) {
```
Comment (bp323) [inline shared/gh/js/views/gh.admin-batch-edit-date.js]: Line of comments

**Comment by bp323** [comment]
Functionality is perfect and good to go, only a few comments on the code remain before merging. Back to @Coenego

File: shared/gh/js/views/gh.admin-batch-edit-date.js (hunk 2 above)
Comment (Coenego) [inline shared/gh/js/views/gh.admin-batch-edit-date.js]: That's only when you add a new day. If you add a new week, OT event's will still be generated.

**Comment by Coenego** [comment]
Back to @bp323

**Comment by bp323** [comment]
Really solid now! Verified and merged.

I've created a ticket for 2 minor issues in the area at https://github.com/CUL-DigitalServices/grasshopper-ui/issues/350

**Pull request merged**